packages/fuels-abigen-macro/tests/harness.rs

```diff
@@ -8,12 +8,20 @@ use fuels_abigen_macro::abigen;
 use fuels_core::constants::NATIVE_ASSET_ID;
 use fuels_core::Token;
 use sha2::{Digest, Sha256};
+use std::env;
 
 /// Note: all the tests and examples below require pre-compiled Sway projects.
 /// To compile these projects, run `cargo run --bin build-test-projects`.
 /// It will build all test projects, creating their respective binaries,
 /// ABI files, and lock files. These are not to be committed to the repository.
 
+fn setup_logger() {
```

Comment:
I feel like the `setup_logger` should be more general to the crate and accessible, so you can use it in the `contract.rs` for instance, instead of the `dbg!` statement.

Reply:
I think the same! I could make us a debugger if there is a need with our macro_rules! and types etc.

Reply:
fn setup_logger() was only to init debugger int test. Now its initialized for every Test.

```rust
#[cfg(test)]
#[ctor::ctor]
fn init() {
    let _ = tracing_subscriber::fmt::try_init();
}
```
You can use for example "event!();" and catch the value that interests you" (event!(Level::DEBUG, "{:?}", receipts);)

Since its TDD, run this on your test like this: "RUST_LOG=fuels_contract::contract=debug cargo test --package fuels-abigen-macro --test harness test_tuples -- --exact --show-output" ----> this is for "test_tuples and my "event" is in "contract" and the flag is "debug"